core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -506,7 +516,16 @@ private[spark] class AppStatusListener(
       if (killedDelta > 0) {
         stage.killedSummary = killedTasksSummary(event.reason, stage.killedSummary)
       }
-      maybeUpdate(stage, now)
+      // Remove stage if there are no active tasks left and stage is already finished
```

Comment:
Comment just repeats the code. You may want to explain when this happens, otherwise the comment is unnecessary.

Reply:
Fixed.